marlowe-test/src/Spec/Marlowe/Plutus/Arbitrary.hs

```diff
@@ -73,7 +83,7 @@ instance Arbitrary Data where
             , do
                 subDataCount <- chooseInt (0, floor $ sqrt @Double $ fromIntegral size)
                 let subDatumSize = size `quot` subDataCount
-                Constr <$> arbitrary <*> vectorOf subDataCount (resize subDatumSize arbitrary)
+                Constr <$> (getNonNegative <$> arbitrary) <*> vectorOf subDataCount (resize subDatumSize arbitrary)
```

Comment:
This fixes a longstanding bug where a constructor could have a non-positive index. As of Plutus v1.15.0.0, the `toAlonzo` data that Cardano API uses now throws an error in such cases.